### ISSUE TITLE
flip returns an int

### ISSUE THREAD

**Issue by georgematheos** [genjax-community/genjax#1445]
```python
key = PRNGKey(0)
x = flip(0.5)(key)
y = flip.mask()(x, 0.5)(key)
```
errors because the value `x` is an int while `mask` requires a boolean flag.

**Comment by sritchie** [comment]
@femtomc , wdyt? should we return a bool from flip? or should we take ints in mask?

**Comment by sritchie** [comment]
@georgematheos , you can do `jnp.bool(x)` in the meantime...

**Comment by femtomc** [comment]
If left to my own devices, I would modify flip to return bools, but I don’t have a strong preference.

having flip return ints allows the return of flip to be used directly to select branches in switch, so changing the behavior does impact that use case.
